common/views/components/IIIFPresentationPreview/IIIFPresentationPreview.js

```diff
@@ -1,199 +1,156 @@
+  return structures.map(structure => {
+    const images = structure.canvases.map(canvasId => {
+      const matchingCanvas =
+        iiifManifest &&
+        iiifManifest.sequences
+          .find(sequence => sequence['@type'] === 'sc:Sequence')
+          .canvases.find(canvas => {
+            return canvas['@id'] === canvasId;
+          });
+      return (
+        matchingCanvas && {
+          orientation:
+            matchingCanvas.thumbnail.service.width >
+            matchingCanvas.thumbnail.service.height
+              ? 'landscape'
+              : 'portrait',
+          uri: iiifImageTemplate(matchingCanvas.thumbnail.service['@id'])({
+            size: '!400,400',
+          }),
+          canvasId,
+        }
+      );
+    });
+    return {
+      label: structure.label,
+      images,
+    };
+  });
+}
 
-  &::-webkit-scrollbar {
-    margin-top: ${props => `${props.theme.spacingUnit}px`};
-    height: ${props => `${props.theme.spacingUnit * 5}px`};
-    background: ${props => props.theme.colors.cream};
-  }
-
-  &::-webkit-scrollbar-thumb {
-    border-radius: 0;
-    border-style: solid;
-    border-width: ${props => `${props.theme.spacingUnit * 2}px 0`};
-    border-color: ${props => props.theme.colors.cream};
-    background: ${props => props.theme.colors.marble};
-  }
+function orderedStructuredImages(structuredImages) {
+  const titlePage = structuredImages.find(
+    structuredImage => structuredImage.label === 'Title Page'
+  );
+  const frontCover = structuredImages.find(
+    structuredImage =>
+      structuredImage.label === 'Front Cover' ||
+      structuredImage.label === 'Cover'
+  );
+  const firstTableOfContents = structuredImages.find(
+    structuredImage => structuredImage.label === 'Table of Contents'
+  );
+  return [titlePage, frontCover, firstTableOfContents].filter(Boolean);
+}
 
-  & > div {
-    display: flex;
-  }
-`;
+function previewThumbnails(
+  iiifManifest = {},
+  structuredImages = [],
+  idealNumber = 5
+) {
+  return structuredImages.length < idealNumber
+    ? structuredImages.concat(
+        randomImages(
+          iiifManifest,
+          structuredImages,
+          idealNumber - structuredImages.length
+        )
+      )
+    : structuredImages;
+}
 
 type Props = {|
-  manifestData: any,
-  showMultiImageWorkPreview: boolean,
+  iiifPresentationLocation: iiifPresentationLocation,
 |};
 
-const IIIFPresentationDisplay = ({
-  manifestData,
-  showMultiImageWorkPreview,
-}: Props) => {
-  const validSequences =
-    (manifestData &&
-      manifestData.sequences
-        // This returns a broken resource
-        .filter(
-          ({ compatibilityHint }) =>
-            compatibilityHint !== 'displayIfContentUnsupported'
-        )) ||
-    [];
-
-  const previewSize = 400;
+const IIIFPresentationDisplay = ({ iiifPresentationLocation }: Props) => {
+  const [imageThumbnails, setImageThumbnails] = useState(false);
```

Comment:
Why false and not just an empty array?
`type Thumbnail[] | boolean` seems a bit hard to fathom down the line.